### PR TITLE
Added delete bucket lifecycle rule capability

### PR DESCRIPTION
**Pull request by bexsoft**
## What does this do?

Adds bucket lifecycle delete rule capability

## How does it look?

Screen Shot 2022-02-10 at 15 28 58
Screen Shot 2022-02-10 at 15 29 01
Screen Shot 2022-02-10 at 15 29 05

Signed-off-by: Benjamin Perez <benjamin@bexsoft.net>

### CODE REVIEW

File: restapi/user_buckets_lifecycle.go [hunk 1]
```diff
@@ -356,3 +365,54 @@ func getEditBucketLifecycleRule(session *models.Principal, params user_api.Updat
...
+	}
+
+	if len(lfcCfg.Rules) == 0 {
+		return errors.New("no rules available to delete")
+	}
+
+	n := 0
+	for _, rule := range lfcCfg.Rules {
+		if rule.ID != params.LifecycleID {
+			lfcCfg.Rules[n] = rule
```
Comment (Alevsk) [inline restapi/user_buckets_lifecycle.go:388]: could `lfcCfg.Rules[n]` be undefined?

**Review by Alevsk — CHANGES_REQUESTED** [review]

File: restapi/user_buckets_lifecycle.go (hunk 1 above)
Comment (bexsoft) [reply restapi/user_buckets_lifecycle.go:388]: In this case no, as we are iterating in the range of Rules list. So if the list is empty, that loop won't be executed

**Review by jinapurapu — APPROVED** [review]

**Review by Alevsk — DISMISSED** [review]

**Comment by jinapurapu** [comment]
I get these errors after deleting the last rule, is that correct
Screenshot from 2022-02-10 15-57-04
Screenshot from 2022-02-10 15-56-50
?

**Comment by bexsoft** [comment]
> I get these errors after deleting the last rule, is that correct Screenshot from 2022-02-10 15-57-04 Screenshot from 2022-02-10 15-56-50 ?

Yes, this error is expected as we are returning 404 when no rules are fetched. you can test this by opening this page in a clean bucket

**Review by dvaldivia — APPROVED** [review]

**Pull request merged**